Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -59,26 +57,30 @@ extension NavigationMapView {
         }
         let currentLegProgress = routeProgress.currentLegProgress
         let currentStepProgress = routeProgress.currentLegProgress.currentStepProgress
+        let currentLegSteps = completeRoutePoints.nestedList[routeProgress.legIndex]
+        var allRemainingPoints = 0
         /**
          Find the count of remaining points in the current step.
          */
-        var allRemainingPoints = getSlicedLinePointsCount(currentLegProgress: currentLegProgress, currentStepProgress: currentStepProgress)
+        let lineString = currentStepProgress.step.shape ?? LineString([])
+        if currentStepProgress.distanceTraveled < 0 {
+            allRemainingPoints += currentLegSteps[currentLegProgress.stepIndex].count
+        } else if let slicedLineString = lineString.trimmed(from: currentStepProgress.distanceTraveled, to: currentStepProgress.step.distance) {
```

Comment:
If I understand correctly `LineString.trimmed(from:to:)` was added in Turf `v2.1.0`, but we did not update Mapbox Directions to it. Should we create new version of Mapbox Directions `v2.1.0-rc.2`, which will use Turf `v2.1.0`? It'll allow to drop explicit Turf usage in Mapbox Navigation.

Reply:
Because right now `MapboxDirections` is not using the new API of `LineString.trimmed(from:to:)` yet of the Turf `v2.1.0`, I'm not sure if we need to upgrade `MapboxDirections` to include this one.

Reply:
It’s OK for MapboxNavigation to explicitly depend on Turf if it has a more specific need than MapboxDirections.